File: output/xCodeEval_8000/Java/C++/43e8b21918fb47b3645b0e1dec805c38.cpp
#include <iostream>
#include <vector>
#include <algorithm>
#include <cmath>
#include <cstring>
#include <map>
#include <set>
#include <queue>
#include <stack>
#include <deque>
#include <bitset>
#include <iomanip>
#include <cassert>
#include <numeric>
#include <functional>
#include <unordered_map>
#include <unordered_set>
using namespace std;

#define fastio ios_base::sync_with_stdio(false);cin.tie(NULL);cout.tie(NULL)
#define pb push_back
#define mp make_pair
#define all(c) (c).begin(), (c).end()
#define sz(x) ((int)(x).size())
#define fi first
#define se second
#define trav(a, x) for(auto& a : x)
#define rep(i, a, b) for(int i = a; i < (b); ++i)
#define per(i, a, b) for(int i = (b)-1; i >= a; --i)
#define endl '\n'
typedef long long ll;
typedef pair<int, int> pii;
typedef vector<int> vi;
typedef vector<ll> vll;
typedef vector<pii> vpii;
typedef vector<vi> vvi;
typedef vector<vll> vvll;
typedef vector<vpii> vvpii;

const int gigamod = 1000000007;

struct FastReader {
    FastReader() {
        ios_base::sync_with_stdio(false);
        cin.tie(NULL);
        cout.tie(NULL);
    }
} fr;

struct CFPS {
    static const int gigamod = 1000000007;
    static int t;
    static double epsilon;
    static bool isPrime[1000001];
    static int smallestFactorOf[1000001];

    static void main() {
        int n, mod;
        cin >> n >> mod;

        vector<ll> dp(n + 1);
        dp[n] = 1;

        for (int i = n - 1; i > 0; i--) {
            dp[i] += dp[i + 1];
            dp[i] %= mod;

            for (ll div = 1; div * i <= n; div++) {
                int lb = div * i;
                int ub = min(div * (i + 1) - 1, (ll)n);
                ll segSum = dp[lb];
                if (ub + 1 <= n)
                    segSum -= dp[ub + 1];
                dp[i] += segSum;
                dp[i] %= mod;
            }

            dp[i] += dp[i + 1];
            dp[i] %= mod;
        }

        cout << (dp[1] - dp[2] + mod) % mod << endl;
    }
};

int CFPS::t = 1;
double CFPS::epsilon = 0.00000001;
bool CFPS::isPrime[1000001];
int CFPS::smallestFactorOf[1000001];

int main() {
    CFPS::main();
    return 0;
}

File: output/GPT3.5-Turbo/xCodeEval_4000/Java/Python/43e8b21918fb47b3645b0e1dec805c38.py
import sys
from typing import List, Tuple, Dict, Any

def main() -> None:
    t = 1
    epsilon = 0.00000001
    isPrime = []
    smallestFactorOf = []

    for _ in range(t):
        n = int(input())
        mod = int(input())

        dp = [0] * (n + 1)
        dp[n] = 1

        for i in range(n - 1, 0, -1):
            dp[i] += dp[i + 1]
            dp[i] %= mod

            for div in range(1, n // i + 1):
                lb = div * i
                ub = min(div * (i + 1) - 1, n)
                segSum = dp[lb]
                if ub + 1 <= n:
                    segSum -= dp[ub + 1]
                dp[i] += segSum
                dp[i] %= mod

            dp[i] += dp[i + 1]
            dp[i] %= mod

        print((dp[1] - dp[2] + mod) % mod)

if __name__ == "__main__":
    main()
